Orange/preprocess/transformation.py

```diff
@@ -48,25 +48,21 @@ def transform(self, c):
         raise NotImplementedError(
             "ColumnTransformations must implement method 'transform'.")
 
-
-class Identity(Transformation):
-    """Return an untransformed value of `c`.
-    """
-    def transform(self, c):
-        return c
-
     def __eq__(self, other):
         return type(other) is type(self) and self.variable == other.variable
```

Comment:
How about the default `__eq__` for `Transformation` along these lines:
```py
return type(other) is type(self) and vars(self) == vars(other)
```

Seems that that is usually what we want and it would make a lot of trivial overloads of eq unnecessary?

`__hash__` is a bit more tricky since class attributes can often be unhashable objects. We could leave it as it is or try hashing a sorted tuple of `vars(self)`. The latter should work for simple cases like `Indicator` and at least fails more noticably if it should have been overloaded in a subclass but was not!

Reply:
I'm hesitant here. It looks like a good idea, but making the parent class to smart might shoot a derived class in the foot. Somebody could add an attribute without realizing it's used in comparisons. Maybe it's better to be explicit.

Reply:
You convinced me at first, but then again, you can make an equivalent claim for the other side:
Somebody could add an attribute without realizing it's _NOT_ used in comparisons.

Maybe I preferred comparing everything (vs the minimum) because it behaves closer to how it was before - if some class doesn't overload eq it might erroneously return False (like up to now), but a True should be correct.

In the end it does not really matter, if something is not working like it should it's a bug to be fixed in either case.
I don't mind merging it with either default.